### ISSUE TITLE
analysing row-col data from cell profiler

### ISSUE THREAD

**Issue by cathalgking** [iSEE/iSEE#521]
I have a question regarding iSEE that I hope that you can help me with. 

Cell profiler (imaging software) outputs several huge datasets (row-column data frames) which take my wet lab colleagues forever to open and choose the columns of interest. Ideally, we would like to be able to pick a few columns from each dataset, put them together in a new dataset and do some basic scatter plots.

How would I do this with iSEE? Which arguments would be best to choose?

I show here an example of a portion of what one of the datasets looks like.
![cpimage](https://user-images.githubusercontent.com/32261323/99959462-f7b89680-2d82-11eb-8934-1cbe312fecc5.png)

**Comment by kevinrue** [comment]
Thanks for your interest in iSEE!

iSEE works on Bioconductor `SummarizedExperiment` objects https://bioconductor.org/packages/release/bioc/html/SummarizedExperiment.html
and sub-classes derived from it, e.g. 
https://bioconductor.org/packages/release/bioc/html/SingleCellExperiment.html

SummarizedExperiment objects can store 3 main types of information:

- matrices of assay data (e.g. gene-by-sample matrix) in the 'assays' slot
- sample (e.g. cell) metadata in the 'colData' slot
- feature (e.g. gene) metadata in the 'rowData' slot

SingleCellExperiment objects are an extension of SummarizedExperiment that can also store:

- matrices of reduced dimension coordinates
 
The key for you is to build one of those objects, storing the relevant information in the relevant slots.
For instance (I never worked with Cell Profiler), from your screenshot, assuming each object is a cell, you could collect relevant columns in your table, and store those columns in the 'colData' slot of a SummarizedExperiment object.
If you have 'features' that are measured for each cell, you may want to create a matrix of features-by-cell and store that matrix in the 'assays' slot.
I would do all that once, and then save the object to a file using `saveRDS(object, "file.rds")`.

Then you'd write a shiny 'app.R' script that loads that object, and calls `iSEE(object)`, possibly with an appropriate configuration script that would control which panels are launched when users start the app and the default values of the different GUI inputs.

We have examples at https://github.com/iSEE/iSEE2018/tree/master/tours
and in recent workshops, e.g. https://github.com/iSEE/iSEEWorkshop2020

I hope that gets you started. There are different steps in the learning curve for iSEE, but to summarise the advice above:

- Create a SummarisedExperiment object.
- Save to disk.
- Write a minimal script that loads the object and launches the iSEE app in default settings.
- Customise the configuration script to control the startup state of the iSEE app.
- Share with your colleagues, e.g. deploy on a Shiny server.

Best wishes
Kevin

**Comment by cathalgking** [comment]
Thanks for your detailed reply.

My colleagues run many experiments per week with Cell Profiler, so, would there be a way to deploy the Shiny app so that my colleagues could just upload the output from Cell Profiler (folder with files - image was an example of 1 file) to the app? Its not really feasible for me to build an object each time. 

Thanks again.

**Comment by kevinrue** [comment]
1) You'll have to think (or perhaps you know already) whether you want one Shiny app per experiment, of one Shiny app that combines all experiments and updates whenever a new one is uploaded.

2) In any case - whether you do it once in advance, or at runtime whenever people launch the app - you'll need to write some code that loads - either the preprocessed object, or the raw files that get processed on the fly - and ends with a call to `iSEE(object)`.
I highly recommend that you look at the 4 'tours' that I linked in my previous message, as examples. In those cases, we separated the creation of the object in the `data.R` script, and the app launch in the `app.R` script. It is entirely possible to combine the two into a single script; you have to choose your own compromise for the amount of time spent on preprocessing in the app launch script vs anything you want to compute in advance.

3) iSEE also offers the possibility to create your own landing page (https://isee.github.io/iSEE/reference/createLandingPage.html), where you can allow users to upload their own data. In that case, you will also need to spend some time yourself, validating and preprocessing those files into a SummarizedExperiment object that iSEE can handle for interactive visualisation.

Bottom line is that there are several ways you can go about this. I would suggest that you start prototyping a simple app with a single experiment, and that you go from there to support additional experiments and/or custom data sets uploaded by your users. In all cases, you will need to spend some time preprocessing the files that you want to support, and configuring iSEE for the best experience of your users.

Your use case is very interesting, and I would be very interested to see the result. If you can develop your prototype publicly and open-source, we could give you feedback along the way.
